pkg/junit/junit.go

```diff
@@ -98,9 +127,24 @@ func parseTestsuite(
 			tc.Duration = duration
 		}
 
+		if testcase.Failure != nil {
+			// Parse owners
+			owners, _, err := parseFailureData(testcase.Failure.Data)
+			if err != nil {
+				l.Warn("Could not parse owners from testcase failure data", "data", testcase.Failure.Data)
+				continue
+			}
+			tc.Owners = owners
+			for _, o := range tc.Owners {
+				allOwners[o] = struct{}{}
+			}
+		}
```

Comment:
LGTM, maybe it would make sense to remove the Owners section from the `testcase.Failure` after we have retrieved the owners part? This way in the point of view of open search the "test_case_status" field won't have "owners" in it.

Reply:
The `testcase.Failure.data` is not encoded into the final opensearch json blobs. As far as I understand the `testcase` variable here is purely input, then we copy the desirable content out into `tc` and only output what is in `tc`. Clean separation means we also don't have to mutate the input.